### DISCUSSION TITLE
Generated PDF files are very large

### DISCUSSION THREAD

**Discussion by normen** [baimard/gestion#302] [Q&A]
Hi,

first of all thanks for this application! I noticed that the generated PDF files become very large for me. A simple quote is about 17MB (!) in size.. My logo is only about 250KB and the text in the PDF can't be that large?

Is there anything that can be done here? Am I doing something wrong? Any guidance would be appreciated.

**Comment by baimard** [comment]
Hello, what's your nextcloud version ?

**Comment by normen** [comment]
Its NextCloud 24

**Comment by infra-monkey** [comment]
Hello,

I have noticed this too.
I'm on nextcloud 25 and the quotes and invoices are about 10MB each for me.
It is not great when sending by email.

It would be very nice for the pdf to be just a few 100k as there is just text and a logo basically.

Is the conversion done on the browser or the server?

**Comment by les-les** [comment]
For an daily use of quote / devis / offert with a full A4 page and the size of 5,0 MB -> 8,3 MB 
it is practical not useable ... adding the offert ant T&C and maybe some other stuff into email ... most of the mailboxes will reject it. 

this might be a showstopper 

testted with NC 25.0.3

**Comment by Bertusand** [comment]
Hi, in my case the resulting PDF weighs around 17MB.
NC 26.0.1 and Gestion version: 2.3.3
Any trick on how to lower the weight of these PDF's?
I'm thinking of making a conversion flow...
Greetings.

**Reply by baimard** [reply]
Maybe your logo file image ? you can reduce size ?

**Reply by Bertusand** [reply]
Thanks for the comment, but I am not using any logo, I have now tried to add a logo in PNG, it generates the PDF correctly but the file size is still too big. 17Mb. :-(

**Reply by infra-monkey** [reply]
@Bertusand On my end, I noticed that the pdf is 2  pages long while there is only one page actually used.
Removing unused pages might help reducing the size.

**Reply by Bertusand** [reply]
Thanks @infra-monkey for the tip, but it's not enough.... :-(
It is something strange, it seems that depending on the screen resolution and client device, the size of the PDF varies by a few MB.

![imatge](https://github.com/baimard/gestion/assets/134603554/78014f18-161b-42b2-b234-58fb92e4d4bd)
It is a pity to have to use a process external to GESTION to have to reduce the weight of the PDF...

**Reply by infra-monkey** [reply]
I have looked a bit at the code and there is nothing strange. So it must be on the jspdf library side.

**Reply by baimard** [reply]
> Thanks @infra-monkey for the tip, but it's not enough.... :-( It is something strange, it seems that depending on the screen resolution and client device, the size of the PDF varies by a few MB.
> 
> ![imatge](https://user-images.githubusercontent.com/134603554/243395031-78014f18-161b-42b2-b234-58fb92e4d4bd.png) It is a pity to have to use a process external to GESTION to have to reduce the weight of the PDF...

Yes it depend on your screen size.

**Comment by infra-monkey** [comment]
@baimard 
Here are some suggestion to improve the size issue.

- convert the html page to png instead of jpg
- create the pdf with the "compress" option
- generate the pdf in text mode instead of image (and that will make the text selectable which is nice to have)

**Reply by baimard** [reply]
I will get to convert in PNG. The size of pdf is a real problem.

Best solution you're right is to turn into html text.

Another solution is to open a page printable like

**Reply by infra-monkey** [reply]
I didn't think about printing to pdf. Indeed a quick test generates a 100k pdf. But for the whole page page and not just the invoice.
So if you can open a page for printing with just the invoice that could work.
Not as clean as generating in text mode but way better.

**Reply by Bertusand** [reply]
In my case, when printing the page and saving it in PDF, it generates a file of about 700kb, with the logo included, which is an acceptable weight to send by mail.

But as @Infra-monkey says, it prints the entire web page and not just the invoice or estimate. 
Then there is the inconvenience that you have to upload the PDF to the NC again.

I think the idea of saving in PDF directly in the NC is very good, comfortable and practical.

And the option to launch a flow script? With ps2pdf from the console I can reduce the PDF from 19Mb to 200Kb:
![imatge](https://github.com/baimard/gestion/assets/134603554/df99b556-fed8-4ab3-80ea-207934325ff6)

**Reply by baimard** [reply]
Be sure that is the next point I will work on
